src/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -0,0 +1,152 @@
+                typeof(DefaultImmutableConverter.ImmutableCreateRangeDelegate<>).MakeGenericType(elementType), null);
+        }
+
+        public override object ImmutableDictionaryCreateRange(Type constructingType, Type elementType)
+        {
+            MethodInfo createRange = ImmutableDictionaryCreateRangeMethod(constructingType, elementType);
+
+            if (createRange == null)
+            {
+                return null;
+            }
+
+            return createRange.CreateDelegate(
+                typeof(DefaultImmutableConverter.ImmutableDictCreateRangeDelegate<,>).MakeGenericType(typeof(string), elementType), null);
+        }
+
+        public override Func<object, TProperty> CreatePropertyGetter<TClass, TProperty>(PropertyInfo propertyInfo) =>
+            (Func<object, TProperty>)CreatePropertyGetter(propertyInfo, typeof(TClass));
+
+        private static Delegate CreatePropertyGetter(PropertyInfo propertyInfo, Type classType)
+        {
+            MethodInfo realMethod = propertyInfo.GetGetMethod();
+            Type objectType = typeof(object);
+
+            var dynamicMethod = new DynamicMethod(
+                realMethod.Name,
+                propertyInfo.PropertyType,
+                new[] { objectType },
+                typeof(ReflectionEmitMemberAccessor).Module,
+                skipVisibility: true);
+
+            ILGenerator generator = dynamicMethod.GetILGenerator();
+
+            generator.Emit(OpCodes.Ldarg_0);
+
+            if (classType.IsValueType)
+            {
+                generator.Emit(OpCodes.Unbox_Any, classType);
+                generator.Emit(OpCodes.Call, realMethod);
+            }
+            else
+            {
+                generator.Emit(OpCodes.Castclass, classType);
+                generator.Emit(OpCodes.Callvirt, realMethod);
+            }
+
+            generator.Emit(OpCodes.Ret);
+
+            return dynamicMethod.CreateDelegate(typeof(Func<,>).MakeGenericType(objectType, propertyInfo.PropertyType));
+        }
+
+        public override Action<object, TProperty> CreatePropertySetter<TClass, TProperty>(PropertyInfo propertyInfo) =>
+            (Action<object, TProperty>)CreatePropertySetter(propertyInfo, typeof(TClass));
+
+        private static Delegate CreatePropertySetter(PropertyInfo propertyInfo, Type classType)
+        {
+            MethodInfo realMethod = propertyInfo.GetSetMethod();
+            Type objectType = typeof(object);
+
+            var dynamicMethod = new DynamicMethod(
+                realMethod.Name,
+                typeof(void),
+                new[] { objectType, propertyInfo.PropertyType },
+                typeof(ReflectionEmitMemberAccessor).Module,
+                skipVisibility: true);
+
+            ILGenerator generator = dynamicMethod.GetILGenerator();
+
+            generator.Emit(OpCodes.Ldarg_0);
+
+            if (classType.IsValueType)
+            {
+                generator.Emit(OpCodes.Unbox_Any, classType);
```

Comment:
Avoid boxing for perf?

Reply:
It doesn't hurt performance because before my changes there were casts in the C# code and all values stored as `object`s in a read stack.

Reply:
That is not true. The only object stored on a read stack is a root object being returned, not POCO properties.